### ISSUE TITLE
baseline models and evaluation script

### ISSUE THREAD

**Issue by seahou3e** [MeiqiGuo/AKBC2021-Abg-CoQA#1]
Hi, I would like to ask if you have plans of releasing the code for baseline models and evaluation scripts? Thanks

**Comment by MeiqiGuo** [comment]
Hi, we provide the GitHub link to baseline code and evaluation scripts in the paper. For our work, we slightly modify their code and we don't plan to release the modified version.

**Comment by seahou3e** [comment]
> Hi, we provide the GitHub link to baseline code and evaluation scripts in the paper. For our work, we slightly modify their code and we don't plan to release the modified version.

Thanks for your reply! So all baseline code and evaluation scripts you refer are from GitHub links in Appendix B ([SogouMRCToolKit](https://github.com/sogou/SogouMRCToolkit), [xlnet_extension_tf](https://github.com/stevezheng23/xlnet_extension_tf) and [Fairseq](https://github.com/pytorch/fairseq))?

**Comment by linqian66** [comment]
> Hi, we provide the GitHub link to baseline code and evaluation scripts in the paper. For our work, we slightly modify their code and we don't plan to release the modified version.

Thanks for the excellent work! Same questions here, can you provide the evaluation script (or link for all sub-tasks) such that the evaluation on my output will be correct?

**Comment by MeiqiGuo** [comment]
> > Hi, we provide the GitHub link to baseline code and evaluation scripts in the paper. For our work, we slightly modify their code and we don't plan to release the modified version.
> 
> Thanks for your reply! So all baseline code and evaluation scripts you refer are from GitHub links in Appendix B ([SogouMRCToolKit](https://github.com/sogou/SogouMRCToolkit), [xlnet_extension_tf](https://github.com/stevezheng23/xlnet_extension_tf) and [Fairseq](https://github.com/pytorch/fairseq))?

Exactly. For the evaluation script, we used the official one for [CoQA](https://stanfordnlp.github.io/coqa/).

**Comment by MeiqiGuo** [comment]
> > Hi, we provide the GitHub link to baseline code and evaluation scripts in the paper. For our work, we slightly modify their code and we don't plan to release the modified version.
> 
> Thanks for the excellent work! Same questions here, can you provide the evaluation script (or link for all sub-tasks) such that the evaluation on my output will be correct?

Sure! For evaluating question answering, we used the official one for CoQA (https://stanfordnlp.github.io/coqa/). For the ambiguity detection (predicting true/false), we used the original F1 score. For generating clarification question, we use the BLEU metric in the nltk package.

**Comment by linqian66** [comment]
Thanks!

**Comment by linqian66** [comment]
@MeiqiGuo For BERT, XLNet and BART mentioned in the paper, do you mean adopting BERT-base (12 layers), XLNet-base (12 layers)? And did you adopt BART-base (6 layers for encoder/decoder)?

**Comment by MeiqiGuo** [comment]
> @MeiqiGuo For BERT, XLNet and BART mentioned in the paper, do you mean adopting BERT-base (12 layers), XLNet-base (12 layers)? And did you adopt BART-base (6 layers for encoder/decoder)?

We have provided details for your questions in Appendix B. We used BERT-base, XLNET-base and BART-large.

**Comment by linqian66** [comment]
> > @MeiqiGuo For BERT, XLNet and BART mentioned in the paper, do you mean adopting BERT-base (12 layers), XLNet-base (12 layers)? And did you adopt BART-base (6 layers for encoder/decoder)?
> 
> We have provided details for your questions in Appendix B. We used BERT-base, XLNET-base and BART-large.

I see, thanks!
Another question regarding the baseline models: for the term "pre-training on CoQA" mentioned in Section 6 & 7.4.2 (BERT+AnsType, XLNet+AnsType, BART), do you mean first fine-tuning on CoQA training set (+validating on dev set), followed by continuing fine-tuning on Abg-CoQA? (since it shows "fine-tune BART on CoQA + Abg-CoQA" in Table 5)

**Comment by linqian66** [comment]
> > @MeiqiGuo For BERT, XLNet and BART mentioned in the paper, do you mean adopting BERT-base (12 layers), XLNet-base (12 layers)? And did you adopt BART-base (6 layers for encoder/decoder)?
> 
> We have provided details for your questions in Appendix B. We used BERT-base, XLNET-base and BART-large.

I see, thanks!
Another question regarding the baseline models: for the term "pre-train on CoQA" (BERT+AnsType, XLNet+AnsType) mentioned in Section 6 (last part, Clarification-based QA) of the paper, does it mean first fine-tuning on CoQA training set (+validating on dev set) as the answer extraction task, followed by continuing fine-tuning on Abg-CoQA?

**Comment by MeiqiGuo** [comment]
> > > @MeiqiGuo For BERT, XLNet and BART mentioned in the paper, do you mean adopting BERT-base (12 layers), XLNet-base (12 layers)? And did you adopt BART-base (6 layers for encoder/decoder)?
> > 
> > 
> > We have provided details for your questions in Appendix B. We used BERT-base, XLNET-base and BART-large.
> 
> I see, thanks! Another question regarding the baseline models: for the term "pre-train on CoQA" (BERT+AnsType, XLNet+AnsType) mentioned in Section 6 (last part, Clarification-based QA) of the paper, does it mean first fine-tuning on CoQA training set (+validating on dev set) as the answer extraction task, followed by continuing fine-tuning on Abg-CoQA?

Right.

**Comment by linqian66** [comment]
For generating the clarification question, did you perform any post-processing/cleaning on model output (keeping the first sentence) and ground-truth answer like `normalize_answer` in CoQA eval [script](https://nlp.stanford.edu/data/coqa/evaluate-v1.0.py), before using nltk BLEU for comparison evaluation?

**Comment by MeiqiGuo** [comment]
> For generating the clarification question, did you perform any post-processing/cleaning on model output (keeping the first sentence) and ground-truth answer like `normalize_answer` in CoQA eval [script](https://nlp.stanford.edu/data/coqa/evaluate-v1.0.py), before using nltk BLEU for comparison evaluation?

Yes, as the last sentence in Appendix B says, we only consider the first generated sentence. We didn't perform any other post-processing.
